src/meta/processors/indexMan/FTIndexProcessor.cpp

```diff
@@ -0,0 +1,172 @@
+    if (!iter->valid()) {
+        LOG(ERROR) << "Edge or Tag could not be found";
+        auto eCode = isEdge
+                  ? nebula::cpp2::ErrorCode::E_EDGE_NOT_FOUND
+                  : nebula::cpp2::ErrorCode::E_TAG_NOT_FOUND;
+        handleErrorCode(eCode);
+        onFinished();
+        return;
+    }
+
+    // verify the columns.
+    auto schema = MetaServiceUtils::parseSchema(iter->val());
+    auto columns = schema.get_columns();
+    for (const auto& col : index.get_fields()) {
+        auto targetCol = std::find_if(columns.begin(), columns.end(), [col] (const auto& c) {
+            return col == c.get_name();
+        });
+        if (targetCol == columns.end()) {
+            LOG(ERROR) << "Column not found : " << col;
+            auto eCode = isEdge
+                       ? nebula::cpp2::ErrorCode::E_EDGE_PROP_NOT_FOUND
+                       : nebula::cpp2::ErrorCode::E_TAG_PROP_NOT_FOUND;
+            handleErrorCode(eCode);
+            onFinished();
+            return;
+        }
+        // Only string or fixed_string types are supported.
+        if (targetCol->get_type().get_type() != meta::cpp2::PropertyType::STRING &&
+            targetCol->get_type().get_type() != meta::cpp2::PropertyType::FIXED_STRING) {
+            LOG(ERROR) << "Column data type error : "
+                       << apache::thrift::util::enumNameSafe(targetCol->get_type().get_type());
+            handleErrorCode(nebula::cpp2::ErrorCode::E_UNSUPPORTED);
+            onFinished();
+            return;
+        }
+        // if the data type is fixed_string,
+        // the data length must be less than MAX_INDEX_TYPE_LENGTH.
+        // else if the data type is string,
+        // will be truncated to MAX_INDEX_TYPE_LENGTH bytes when data insert.
+        if (targetCol->get_type().get_type() == meta::cpp2::PropertyType::FIXED_STRING &&
+            *targetCol->get_type().get_type_length() > MAX_INDEX_TYPE_LENGTH) {
+            LOG(ERROR) << "Unsupported data length more than "
+                       << MAX_INDEX_TYPE_LENGTH << " bytes : "
+                       << col << "(" << *targetCol->get_type().get_type_length() << ")";
+            handleErrorCode(nebula::cpp2::ErrorCode::E_UNSUPPORTED);
+            onFinished();
+            return;
+        }
+    }
+
+    // Check fulltext index exist.
+    auto ret = doPrefix(MetaServiceUtils::fulltextIndexPrefix());
+    if (!nebula::ok(ret)) {
+        auto retCode = nebula::error(ret);
+        LOG(ERROR) << "Fulltext Prefix failed, "
+                   << apache::thrift::util::enumNameSafe(retCode);
+        handleErrorCode(retCode);
+        onFinished();
+        return;
+    }
+    auto it = nebula::value(ret).get();
+    // If already have a full-text index that depend on this tag or edge,  reject to create it.
+    // even though it's a different column.
+    while (it->valid()) {
+        auto indexName = MetaServiceUtils::parsefulltextIndexName(it->key());
```

Comment:
If we forbid a schema has more than one FTIndex, the name of FTIndex seems meaningless? Why not just use tag name or edge name?

BTW, consider change the logic from `CreateFTIndex` to `add one column to FTIndex`. IMO, the forbid here is a little aggressive, and it will be hard to extend later. For example, if we want to add another column, we could only drop then recreate. Actually, we only need to rebuild FTIndex of this column.

Reply:
> If we forbid a schema has more than one FTIndex, the name of FTIndex seems meaningless? Why not just use tag name or edge name?
> 
Good question, I think it's more flexible of this way. The users can name it to `tag name` or `edge name`,  also named another. either way, we can show them through `SHOW FULLTEXT INDEXES`. 
```
(root@nebula) [ft_test_space]> show fulltext indexes
+-----------------+-------------+-------------+----------------------+
| Name            | Schema Type | Schema Name | Fields               |
+-----------------+-------------+-------------+----------------------+
| "nebula_ft_idx" | "Tag"       | "player"    | "c_fixed_string_256" |
+-----------------+-------------+-------------+----------------------+
```

> BTW, consider change the logic from `CreateFTIndex` to `add columns to FTIndex`. IMO, the forbid here is a little aggressive, and it will be hard to extend later. For example, if we want to add another column, we could only drop then recreate. Actually, we only need to rebuild FTIndex of this column.
two ways : 1, drop ftindex -> recreate ftindex -> rebuild ftindex; 2, alter ftindex add column -> rebuild ftindex. Actually I prefer the second one. but the current implementation is the first one, the reason is that we don't support `REBUILD` yet. I think we can do it after `REBUILD` ready. WDYT？

Reply:
>  I think we can do it after REBUILD ready. WDYT？
Could consider implement `add columns to FTIndex` here, it won't conflict with this PR. Just forbid upcoming alter. Could do it later as well.

Reply:
OK , Let's do that in the next PR, the current PR is too big.